src/test/kotlin/com/nftco/flow/sdk/cadence/fields/JsonCadenceBuilderAddressFieldTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.nftco.flow.sdk.cadence.fields
+
+import com.nftco.flow.sdk.cadence.AddressField
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotEquals
+import org.junit.jupiter.api.Test
+
+class JsonCadenceBuilderAddressFieldTest {
+
+    @Test
+    fun `Test equality for AddressField with the same value`() {
+        val address1 = AddressField("0x123")
+        val address2 = AddressField("0x123")
+        assertEquals(address1, address2)
+        assertEquals(address1.hashCode(), address2.hashCode())
+    }
+
+    @Test
+    fun `Test equality for AddressField with different values`() {
+        val address1 = AddressField("0x123")
+        val address2 = AddressField("0x456")
+        assertNotEquals(address1, address2)
+        assertNotEquals(address1.hashCode(), address2.hashCode())
+    }
+
+    @Test
+    fun `Test equality for AddressField with lowercase value`() { // need to investigate this case
```

Comment:
Need to investigate

Reply:
@franklywatson do you by chance know if the SDK should handle case-insensitive wallet addresses? Currently both the address value and hashCode() are case-sensitive, so this case fails.

Reply:
I think it should thanks